### ISSUE TITLE
Damselfly is not compatible with SSL

### ISSUE THREAD

**Issue by asitemade4u** [Webreaper/Damselfly#322]
1. I wish to be able to connect securely to our self-hosted version of Damselfly from another computer/another location
2. We are currently allowing that on most of our self-hosted applications using Nginx Proxy Manager (NPM) and Cloudflare, with good results
3. However Damselfly does not seem to accommodate it so well: when accessing it through the new htps address, it shows an inert GUI on which most of the elements are missing.

Of course, as allowed by NPM, I have disabled:

- asset caching
- HSTS
- HTTP/2 support

but to no avail.

Please help.
Best,
Stephen

**Comment by Webreaper** [comment]
This may be tricky for me to debug, as I have no way of reproducing the issue. Can you please:

- check the browser console log and see if there's any errors etc that might be helpful.
- attach or email (info@damselfly.info) a full log of Damselfly running and being accessed like this (logs are in the /config/logs folder).

I access damselfly via a VPN and it works just fine. But I'll do some research. My hunch is that the proxy etc is interfering with the signalR connection.

**Comment by Webreaper** [comment]
It looks from the error you emailed that web sockets are blocked. Have you definitely enabled them?

This thread might help? 

Nginx Proxy Manager Not Passing WebSocket - https://www.reddit.com/r/selfhosted/comments/nkxfal/nginx_proxy_manager_not_passing_websocket/

**Comment by asitemade4u** [comment]
Yup, just checked + I mimicked the settings described in the post + used 
another browser which had never connected to Damselfly > the same

**Comment by davidfowl** [comment]
Sticky sessions are required to make it work.

**Comment by Webreaper** [comment]
Thanks @davidfowl. 

@asitemade4u can you try that and see if it helps?

**Comment by UhtredTheBold** [comment]
Just to say I've got damselfly working fine through nginx and SSL without doing anything I would consider unusual.

I've not used NPM before so I don't know how useful this is, but here are the options in my connection block.

       `proxy_pass http://xxx.xxx.xxx.xxx:6363;

        proxy_set_header Host $host;

        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

        proxy_set_header Upgrade $http_upgrade;

        proxy_set_header Connection $connection_upgrade;

        proxy_set_header X-Forwarded-Proto $scheme;`
